### ISSUE TITLE
[HybridApp] Launch the NewDot experience on HybridApp for new sign-ups on native mobile

### ISSUE THREAD

**Issue by trjExpensify** [Expensify/App#45548]
### Problem

Coming from [this thread](https://expensify.slack.com/archives/C036QM0SLJK/p1720533495933469?thread_ts=1719577680.548439&cid=C036QM0SLJK) and [this one](https://expensify.slack.com/archives/C036QM0SLJK/p1720725205290589?thread_ts=1720723258.872159&cid=C036QM0SLJK). 

The goal is to put 100% of new sign-ups through NewDot with the August release. Plans are a-foot with the new home page on expensify.com to redirect those sign-ups to new.expensify.com. A large portion of sign-ups come from the native mobile app though, so we'll be sending all of those to the OldDot experience on HybridApp, as the new home page logic doesn't cover them. 

In order to not miss out on that large population of new sign-ups on native mobile, we want to load the newDot experience in HybridApp for them.

### Solution

We'll need to either pass the `shouldSetClassicRedirectNVP` param from Mobile, or default to setting it on the backend with some additional handling for the new web landing page logic. The first solution is simpler but requires a Hybrid App deploy.

Note: I don't know how closely correlated this is with the [other issue](https://github.com/Expensify/App/issues/45015) Jules, like if we need to hold on it or if SWM can work on this issue, but wanted to make sure we have an issue for it to verify as you mentioned in the thread. 

CC: @AndrewGable

**Comment by Julesssss** [comment]
Focusing on other hybrid tasks

**Comment by trjExpensify** [comment]
We're still rolling out to existing installs before putting this in place, Melv.

**Comment by Julesssss** [comment]
Still not being picked up

**Comment by trjExpensify** [comment]
We shouldn't be that far off being able to get this one in place, Melv.

**Comment by AndrewGable** [comment]
I made a web PR, but we are showing the "Welcome" modal via `completedHybridAppOnboarding`.

I am going to see how we can hide this modal for new users going forward:


https://github.com/user-attachments/assets/28b22535-9757-4bfd-8efb-67cae6d82f1a

**Comment by AndrewGable** [comment]
Ok I got this working with a small auth change 🎉 


https://github.com/user-attachments/assets/f254c92f-c0f9-4f25-97b7-2404d0065310

I need to write some tests on Monday, but feel free to review the changes @Julesssss

**Comment by AndrewGable** [comment]
FYI auth PR was reverted

**Comment by Julesssss** [comment]
Both PRs merged, awaiting deployment and testing.

**Comment by AndrewGable** [comment]
We ran into some issues during testing this, but have resolved them in the latest TestFlight. We will need another production deploy in order to start rolling this out to users.

**Comment by trjExpensify** [comment]
Thread this week, [here](https://expensify.slack.com/archives/C07HPDRELLD/p1729590307243149?thread_ts=1729565763.405919&cid=C07HPDRELLD).

**Comment by AndrewGable** [comment]
I think hopefully one more releaseœ

**Comment by Julesssss** [comment]
Here's the 'final' release: https://github.com/Expensify/Expensify/issues/442751

**Comment by trjExpensify** [comment]
@AndrewGable we can release the beta now, right? We fixed the remaining NewDot bugs on `9.0.58-2` didn't we: https://github.com/Expensify/Expensify/issues/442751

**Comment by AndrewGable** [comment]
I believe we can, do we want to do one last test before turning it on?

**Comment by trjExpensify** [comment]
I just downloaded the iOS app from the production app store, and the receipt scanning thumbnail doesn't go blank anymore on this version: 

https://github.com/user-attachments/assets/cf989b1e-7399-488c-bf4c-544b7cc3ef8e

My trj.chat domain is on the beta if anyone wants to create an account with `tom+<insertsomething>@trj.chat` to test.

**Comment by AndrewGable** [comment]
I think we are really close here but @Julesssss has been leading this recently

**Comment by Julesssss** [comment]
I'm struggling to find the thread, but we discussed with @trjExpensify last week and planned to hold on the 'redirect admins back to OldApp when they indicate they have 11+ member workspaces' changes.

**Comment by trjExpensify** [comment]
[This one](https://expensify.slack.com/archives/C07HPDRELLD/p1733484433968479). Issue [here](https://github.com/Expensify/App/issues/53906).

**Comment by trjExpensify** [comment]
@Julesssss @AndrewGable just confirming the plan of action for the final steps here. 

- Merge & Deploy @mateuuszzzzz [PR](https://github.com/Expensify/App/pull/54033). 
- `*` the `hybridAppRedirect` beta to put all new sign-ups in the NewDot experience on sign-up.

**Comment by Julesssss** [comment]
@trjExpensify sounds good. I'm not sure when I'll be able to find time to verify this though. Were you planning to ship ASAP?

**Comment by trjExpensify** [comment]
Yeah, with the NewDot app listing deprecated looking to ship it ASAP.

**Comment by trjExpensify** [comment]
> @Julesssss @AndrewGable just confirming the plan of action for the final steps here.
> 
> * Merge & Deploy @mateuuszzzzz [PR](https://github.com/Expensify/App/pull/54033).
> * `*` the `hybridAppRedirect` beta to put all new sign-ups in the NewDot experience on sign-up.

I've asked in that PR for the 11+ redirect to fix the conflicts, so we can merge it.

**Comment by trjExpensify** [comment]
Update: 

- [Main PR](https://github.com/Expensify/App/pull/54033) for the [11+ redirect to Classic issue](https://github.com/Expensify/App/issues/53906) is on prod. 
- A bug was reported with that [here](https://github.com/Expensify/App/issues/55167) which causes multiple workspaces to be created if you navigate back and forth and complete the employee count step multiple times. [The PR](https://github.com/Expensify/App/pull/55168) for that is merged, and we're discussing [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1736801404028519?thread_ts=1736772274.621189&cid=C01GTK53T8Q) about CP'ing it to staging, so it can go out to prod with today's deploy. I don't feel like that's a blocker for our HybridApp Redirect beta release.
- Awaiting input from @mateuuszzzzz on the [NewDot Web side of the 11+ redirect issue](https://github.com/Expensify/App/issues/53906#issuecomment-2578958107). Though I don't think that's a blocker for @AndrewGable to pull the trigger on releasing the `hybridAppRedirect` beta either.

**Comment by trjExpensify** [comment]
PR for NewDot web 11+ redirects is [in progress](https://github.com/Expensify/App/pull/55514#issuecomment-2607194321).

**Comment by trjExpensify** [comment]
Jules is working on the Web PR to unblock that App PR linked above.

**Comment by AndrewGable** [comment]
Not working on this one so un-assigning for now, happy to review any code though!

**Comment by trjExpensify** [comment]
Update: 

- We've resolved the string/boolean bug. 
- App unfortunately hit a couple of crashes on sign-up this week. Fixes for those are on prod, but we're waiting for the next release to get to the App Store before we can add the beta for `gmail.com` and then full release to `*` for all.

**Comment by trjExpensify** [comment]
We're live on prod for all new sign-ups! 🎉 @Julesssss confirming we can close this? I'm not 100% sure what that Web-E Draft PR has to do with this one? 🤔

**Comment by Julesssss** [comment]
> We're live on prod for all new sign-ups! 🎉 [@Julesssss](https://github.com/Julesssss) confirming we can close this? I'm not 100% sure what that Web-E Draft PR has to do with this one? 🤔

Oops yeah I psted the wrong URL in that PR description. Updated. 

Yes lets close this out. We got there eventually!

**Comment by trjExpensify** [comment]
Haha, woop. Solid work!
